tests/nlu/selectors/test_selectors.py

```diff
@@ -427,29 +544,42 @@ async def test_margin_loss_is_not_normalized(
     ],
 )
 async def test_softmax_ranking(
-    component_builder: ComponentBuilder,
-    tmp_path: Path,
     classifier_params: Dict[Text, int],
     data_path: Text,
     output_length: int,
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
 ):
-    pipeline = as_pipeline(
-        "WhitespaceTokenizer", "CountVectorsFeaturizer", "ResponseSelector"
-    )
-    assert pipeline[2]["name"] == "ResponseSelector"
-    pipeline[2].update(classifier_params)
+    pipeline = [
+        {"name": "WhitespaceTokenizer"},
+        {"name": "CountVectorsFeaturizer"},
+    ]
+    loaded_pipeline = [
+        registry.get_component_class(component.pop("name"))(component)
+        for component in copy.deepcopy(pipeline)
+    ]
 
-    _config = RasaNLUModelConfig({"pipeline": pipeline})
-    (trained_model, _, persisted_path) = await rasa.nlu.train.train(
-        _config,
-        path=str(tmp_path),
-        data=data_path,
-        component_builder=component_builder,
+    importer = RasaFileImporter(training_data_paths=["data/test_selectors"])
+    training_data = importer.get_nlu_data()
+
+    for component in loaded_pipeline:
+        component.train(training_data)
```

Comment:
this bit of code is used multiple times in the tests. How about extracting it into a helper function / fixture?

Reply:
Yup, I actually just tidied up the tests this morning by extracting 2 helper methods, have just pushed a few minutes ago 👍🏼